engine/hash_table.cpp

```diff
@@ -178,10 +181,11 @@ Status HashTable::SearchForWrite(const KeyHashHint &hint,
   if (found) {
     (*entry_base)->header.status = HashEntryStatus::Updating;
   } else {
-    if ((*entry_base) == reusable_entry) {
-      if ((*entry_base)->header.status == HashEntryStatus::CleanReusable) {
-        (*entry_base)->header.status = HashEntryStatus::Updating;
+    if (reusable_entry) {
```

Comment:
reusable_entry is not null doesn't mean it is reused, we only reuse it while no spare entry in existing bucket, so don't modify this